mono/metadata/object.c

```diff
@@ -272,24 +273,25 @@ void
 mono_runtime_class_init (MonoVTable *vtable)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
+	MonoError error;
 
-	mono_runtime_class_init_full (vtable, TRUE);
+	mono_runtime_class_init_full (vtable, &error);
+	mono_error_assert_ok (&error);
```

Comment:
The old behavior was to call `mono_runtime_class_init_full(vtable, TRUE)` which meant raise on error.  Which means that if it ever failed, the API client would die.  It seemed better that the new behavior would be to die faster rather than silently succeed without reporting anything.